### DISCUSSION TITLE
userSegment api is not working

### DISCUSSION THREAD

**Discussion by Nik720** [Samagra-Development/SamagraX-community#2] [Troubleshooting and Clarifications] [answered]
@SaketSinha : Can you please check below APIs are failing

**{{base_url}}/admin/v1/userSegment/all** 
API is giving an error,  504 Gateway Time-out error. 

**{{base_url}}/admin/v1/userSegment/create** with below request body
`{
    "data": {
        "name": "SamagraX-Team-Users-test",
        "all": {
            "type": "gql",
            "config": {
                "pageParam": "page",
                "cadence": {
                    "concurrent": true,
                    "pagination": false,
                    "perPage": 10000,
                    "retries": 0,
                    "retries-interval": 10,
                    "timeout": 60
                },
                "credentials": {
                    "variable": "commsgql",
                    "vault": "samagra"
                },
                "gql": "query getMissionPrernaUsers {users: getUsersByApplication(application: \"SamagraX Testing App\", limit: 2000) {id full_name mobilePhone data: jdata}}"
            }
        },
        "byID": {
            "type": "gql",
            "config": {
                "pageParam": "page",
                "cadence": {
                    "concurrent": true,
                    "pagination": false,
                    "perPage": 10000,
                    "retries": 0,
                    "retries-interval": 10,
                    "timeout": 60
                },
                "verificationParams": {
                    "id": "f5c49bcc-b96c-4206-950e-a2ba4319f149"
                },
                "credentials": {
                    "variable": "commsgql",
                    "vault": "samagra"
                },
                "gql": "query Query($id: String) {users: getUsersByQuery(queryString: $id) {id username mobilePhone data: jdata}}"
            }
        },
        "byPhone": {
            "type": "gql",
            "config": {
                "pageParam": "page",
                "cadence": {
                    "concurrent": true,
                    "pagination": false,
                    "perPage": 10000,
                    "retries": 0,
                    "retries-interval": 10,
                    "timeout": 60
                },
                "verificationParams": {
                    "phoneNo": "9415787824"
                },
                "credentials": {
                    "variable": "commsgql",
                    "vault": "samagra"
                },
                "gql": "query Query($phoneNo: String) {users: getUsersByQuery(queryString: $phoneNo) {id username mobilePhone data: jdata}}"
            }
        }
    }
}`

 is giving an error, "UserSegment could not be registered. Services down."
 
 referring to postman collection: https://documenter.getpostman.com/view/7043186/Tz5qaxaN#e0f3d524-636f-4dd2-9f14-51d3fe2e221a

**Comment by SaketSinha** [comment, marked as answer]
Hi @Nik720 , acknowledging that these API's to be not working at the moment:
1. {{base_url}}/admin/v1/userSegment/all - Time Out
2. {{base_url}}/admin/v1/userSegment/create -  200 Ok - With error message , "UserSegment could not be registered. Services down."

These will be fixed by tomorrow.

**Reply by SaketSinha** [reply]
The APIs are now working. Please check and confirm
